packages/wdio-browserstack-service/src/insights-handler.ts

```diff
@@ -0,0 +1,421 @@
+            testMetaData['feature'] = {
+                path: gherkinDocument.uri,
+                name: featureData.name,
+                description: featureData.description,
+            }
+        }
+
+        this._tests[uniqueId] = testMetaData
+
+        await this.sendTestRunEventForCucumber(world, 'TestRunStarted')
+    }
+
+    async afterScenario (world: ITestCaseHookParameter) {
+        await this.sendTestRunEventForCucumber(world, 'TestRunFinished')
+    }
+
+    async beforeStep (step: Frameworks.PickleStep, scenario: Pickle) {
+        const uniqueId = getUniqueIdentifierForCucumber({ pickle: scenario } as ITestCaseHookParameter)
+        let testMetaData = this._tests[uniqueId]
+        if (!testMetaData) {
+            testMetaData = {
+                steps: []
+            }
+        }
+
+        if (testMetaData && !testMetaData['steps']) {
+            testMetaData['steps'] = []
+        }
+
+        testMetaData['steps']?.push({
+            id: step.id,
+            text: step.text,
+            keyword: step.keyword,
+            started_at: (new Date()).toISOString()
+        })
+
+        this._tests[uniqueId] = testMetaData
+    }
+
+    async afterStep (step: Frameworks.PickleStep, scenario: Pickle, result: Frameworks.PickleResult) {
+        const uniqueId = getUniqueIdentifierForCucumber({ pickle: scenario } as ITestCaseHookParameter)
+        let testMetaData = this._tests[uniqueId]
+        if (!testMetaData) {
+            testMetaData = {
+                steps: []
+            }
+        }
+
+        if (testMetaData && !testMetaData['steps']) {
+            testMetaData['steps'] = []
+            testMetaData['steps'].push({
+                id: step.id,
+                text: step.text,
+                keyword: step.keyword,
+                finished_at: (new Date()).toISOString(),
+                result: result.passed ? 'PASSED' : 'FAILED',
+                duration: result.duration,
+                failure: result.error ? removeAnsiColors(result.error) : result.error
+            })
+        } else if (testMetaData){
+            let stepDetails = testMetaData['steps']?.find(item => item.id == step.id)
+            if (stepDetails) {
+                stepDetails.finished_at = (new Date()).toISOString()
+                stepDetails.result = result.passed ? 'PASSED' : 'FAILED'
+                stepDetails.duration = result.duration
+                stepDetails.failure = result.error ? removeAnsiColors(result.error) : result.error
+            }
+        }
+
+        this._tests[uniqueId] = testMetaData
+    }
+
+    // misc methods
+
+    async browserCommand (commandType: string, args: BeforeCommandArgs & AfterCommandArgs, test?: Frameworks.Test | ITestCaseHookParameter) {
+        if (commandType == 'client:beforeCommand') {
+            this._commands[`${args.sessionId}_${args.method}_${args.endpoint}`] = args
+        } else {
+            if (test == undefined) return
```

Comment:
Do an early return so we don't need the else and indentation
```suggestion
        if (commandType == 'client:beforeCommand') {
            this._commands[`${args.sessionId}_${args.method}_${args.endpoint}`] = args
            return
        }
        
        if (!test) {
            return
        }
```